core/src/test/java/org/apache/iceberg/jdbc/TestJdbcCatalog.java

```diff
@@ -685,6 +685,29 @@ public void testRenameTable() {
         .hasMessage("Table already exists: db.tbl2-newtable");
   }
 
+  @Test
+  public void testCreateTableAfterRename() {
```

Comment:
what about adding such a test to `CatalogTests`?

Reply:
I did it, it works wel lfor Hive, InMemory, Jdbc and Nessie catalogs. But in tests for REST catalog I'm not sure how to proceed. So far I configured property `unique_table_location` for REST catalog **backend**. For me it seems valid option, backend decides about default table location. But maybe we want to handle it differently in REST catalog, anyway I think it worh another PR.